### PR TITLE
Wasm: Import functions in the indirect table

### PR DESCRIPTION
**Pull request by Cellule**
Support imported functions in the indirect table.
Create an internal function to transition from wasm to js.
Some refactor in the WasmModule

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/1763)

### CODE REVIEW

**Comment by Cellule** [comment]
@MikeHolman @arunetm @Krovatkin Can you guys review this please ?

File: lib/WasmReader/WasmBinaryReader.cpp [hunk 1]
```diff
@@ -646,15 +652,15 @@ void WasmBinaryReader::ReadExportTable()
         {
         case ImportKinds::Function:
         {
-            FunctionIndexTypes::Type type = m_module->GetFunctionIndexType(index);
-            if (type == FunctionIndexTypes::Invalid)
+            uint32 normIndex = 0;
+            FunctionIndexTypes::Type type = m_module->GetFunctionIndexType(index, &normIndex);
+            if (type != FunctionIndexTypes::Import && type != FunctionIndexTypes::Function)
```
Comment (arunetm) [inline lib/WasmReader/WasmBinaryReader.cpp]: Cant we check for Invalid?

File: lib/WasmReader/WasmCustomReader.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return wbEnd;
+}
+
+void WasmCustomReader::AddNode(WasmNode node)
+{
+    m_nodes.Add(node);
+}
+
+};
+#endif
```
Comment (arunetm) [inline lib/WasmReader/WasmCustomReader.cpp]: nit:  newline needed

**Comment by Cellule** [comment]
Closing to retarget against master

**Pull request closed without merging**